src/foam/dao/CSVSink.js

```diff
@@ -76,34 +127,95 @@ foam.CLASS({
             Null: function(value) {}
           }, function(value) {
             this.output_(value.toString());
-          })
+        }),
+      javaCode: `
+        switch (value) {
+          case value == null: 
+            break;
+          case value instanceof String:
+            setCsv(getCsv() +  value.replace("\\"", "\\"\\""));
+            break;
+          case value instanceof Number:
+            setCsv(getCsv() + value.toString());
+            break;
+          case value instanceof Boolean: 
+            setCsv(getCsv() + value.toString());
+            break;
+          case value instanceof Date:
+            output_(value.toDateString());
+            break;
+          case value instanceof Array:
+            output_(foam.json.Pretty.stringify(value));
```

Comment:
The Pretty outputter can add newlines, which would break the CSV format, most CSV handlers don't allow newlines.